moby/moby
# Start containers in their own cgroup namespaces

**- What I did**

This adds [cgroup namespace support](http://man7.org/linux/man-pages/man7/cgroup_namespaces.7.html) to docker. It's enabled for all containers that are not run with `--privileged`, if the kernel supports it.

Fixes #38332.

**- How I did it**

Now that [runc supports cgroup namespaces](https://github.com/opencontainers/runc/pull/1916), just add a namespace of type "cgroup" to the list.

At daemon startup, this looks for `/proc/self/ns/cgroup` to be present to determine whether or not to enable it.

**- How to verify it**

Run a new container and verify that its pid 1 is running in its own cgroup namespace.

**- Description for the changelog**

Add cgroup namespace support for Linux containers

**- A picture of a cute animal (not mandatory but encouraged)**
